lib/library.hoon

```diff
@@ -0,0 +1,135 @@
+/-  *post, resource, library, spider
+/+  *graph-store, strandio
+|%
+++  incr-index
+  :: increments the last value in the index by 1
+  |=  [=index:post]
+  ^-  index:post
+  =/  last=@                (sub (lent index) 1)  :: calculate the array index of the last item
+  =/  old-revision-count=@  (snag last index)     :: get the last item
+  (snap index last (add 1 old-revision-count))    :: replace the value of last item with 1 added to it
+++  scry-for
+  |*  [=mold =path]
+  .^  mold
+    %gx
+    (scot %p our.bowl)
+    %library-proxy
+    (scot %da now.bowl)
+    (snoc `^path`path %noun)
+  ==
+::
+++  make-meta-contents
+  |=  [=book:library]
+  ^-  (list content)
+  ~[[%text (crip title.book)] [%text (crip isbn.book)]]
+++  make-comment-contents
+  |=  [comment-text=comment:library]
+  ^-  (list content)
+  ~[[%text (crip comment-text)]]
+::
+++  is-allowed
+  |=  [requester=ship host=ship =policy:library]
+  ^-  ?
+  ?:  =(requester host)  :: host is always allowed
+    %.y
+  ?-  -.policy
+      %open       %.y
+      %children   (team:title host requester)
+      %whitelist  (~(has in ships.policy) requester)
+  ==
+++  index-to-path
+  |=  [idx=index:post]
+  ^-  path
+  (turn idx (cury scot %ud))
+::
+::
+++  create-library-update
+  |=  [[owner=ship name=@tas] time-sent=time]
+  ^-  update
+  :-  time-sent
```

Comment:
```
:^  time-sent  %add-graph
  [owner name]
[*graph `%graph-validator-library %.n]
```

same thing, just play around with the style and see what you like best!

Reply:
https://github.com/ynx0/library/pull/3/commits/e0e660826361bf85fd8beda15a64217c9fc82b35

Reply:
https://github.com/ynx0/library/pull/3/commits/86f60a7f7ed52771d7bbc945b9557a397b810224